dask/base.py

```diff
@@ -416,14 +416,32 @@ def tokenize(*args, **kwargs):
     return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
 
 
+def dont_optimize(dsk, keys):
+    return dsk
+
+
+def optimization_function(obj):
+    if isinstance(obj, type):
+        cls = obj
+    else:
+        cls = type(obj)
+    name = cls.__name__.lower() + '_optimize' # dask.set_options(array_optimize=foo)
```

Comment:
I agree that this feels odd.  However so too does the following:

```python
class Array(Base):
    optimization_name = 'array'
```

Can you recommend anything more clean?